tests/Resources/Page/OCPDashboard/Pods/Pods.robot

```diff
@@ -51,3 +51,38 @@ Get Pod Status
     ${data}       Run Keyword   OpenShiftCLI.Get   kind=Pod
     ...    namespace=${namespace}   label_selector=${label_selector}
     [Return]      ${data[0]['status']['phase']}
+
+Get POD Names
+    [Documentation]    Get the name of list based on
+    ...    namespace and label selector and return the
+    ...    name of all the pod with matching label selector
+    [Arguments]   ${namespace}   ${label_selector}
+    ${pod_name}    Create List
+    ${status}      Check If POD Exists       ${namespace}        ${label_selector}
+    IF    '${status}'=='PASS'
+         ${data}        OpenShiftCLI.Get   kind=Pod     namespace=${namespace}   label_selector=${label_selector}
+         FOR    ${index}    ${element}    IN ENUMERATE    @{data}
+                Append To List    ${pod_name}     ${data[${index}]['metadata']['name']}
+         END
+    ELSE
+         FAIL    No POD found with the provided label selector in a given namespace '${namespace}'
+    END
+    [Return]    ${pod_name}
+
+Get Container Restart Counts
+    [Documentation]    Get the container name with restart
+    ...    count for each pod provided
+    [Arguments]        ${name}   ${namespace}
+    ${restart_c}      Create Dictionary
+    FOR    ${element}    IN    @{name}
+        ${c_detail}    Create Dictionary
+        ${data}    OpenShiftCLI.Get   kind=Pod     namespace=${namespace}   field_selector=metadata.name==${element}
+        FOR    ${index}    ${container}    IN ENUMERATE    @{data[0]['status']['containerStatuses']}
+               ${value}    Convert To Integer    ${container['restartCount']}
+               IF    ${value} > ${0}
```

Comment:
why not setting the container in the dictionary even when the restart count is 0 ? The keyword says to return the restart count for each container of each given pods, should I think it should return even 0 as value. The filtering can be done outside the keyword

Reply:
I can see the confusion,will change the keyword name to _Get Containers ​With Non Zero Restart Counts_.The reason we don't want to capture is ) restart container is by default restart count is zero.IMO another keyword to filter it out is not worth it. Let me know if it is okay?

Reply:
you already have the keyword to filter them out, without creating a new one: `Verify Restart Container Verification`

Reply:
If you look at that **Verify Restart Container Verification** it calls this keyword and just check the length because my filter is actually happening here

Reply:
i have change the name and kept it as it is

Reply:
> If you look at that Verify Restart Container Verification it calls this keyword and just check the length because my filter is actually happening here

yeah I saw it. Was suggesting to move your filter there. However, never mind, now it's better with the new title